src/NHibernate.Test/Async/NHSpecificTest/SqlConverterAndMultiQuery/Fixture.cs

```diff
@@ -27,9 +28,12 @@ protected override void Configure(Configuration configuration)
 
 		protected override bool AppliesTo(Dialect.Dialect dialect)
 		{
-			// MsSqlCe throws InvalidOperationException instead of a DbException for these tests, preventing
+			// If prepare_sql is enabled, with most databases prepare will fail with an InvalidOperationException,
+			// turned into ADOException by prepare code and then not passed to the exception converter.
+			// MsSqlCe throws InvalidOperationException during Prepare (even when disabled), preventing
 			// the test SqlConverter to do its job.
-			return !(Dialect is MsSqlCeDialect);
+			var prepareSql = PropertiesHelper.GetBoolean(Environment.PrepareSql, cfg.Properties, false);
+			return !(prepareSql || Dialect is MsSqlCeDialect);
```

Comment:
Maybe it shall be only for SqlClientDriver?

Reply:
`return !(prepareSql && !(driver is SqlClientDriver) || Dialect is MsSqlCeDialect);` ? Or just `return driver is SqlClientDriver` ?

Reply:
Second

Reply:
I do not really get why it should be considered a Sql Server only test. For me it is more like this tests reveal there is some "too wide" catching in place causing the feature to be unavailable for most databases when prepare is enabled. Anyways, that is a quite secondary feature in my mind, and the code lacks indications as to why this "wide" catch is there, so I do not think we should investigate/fix this point, at least not for 5.0.

Reply:
Ok, leave it as is

Reply:
Or maybe more accurately, reveals that the test implementation (closing the connection) is not suitable for most databases when prepare is enabled. Maybe should it instead use a trick (like the nullable no null column in some transactions tests for failures) to alter the schema and use that to cause an actual db side failure.

Reply:
Well, already changed, with comment corresponding to my last comment just above.

Reply:
Need to regen the async

Reply:
Done. Thanks for the reminder.